### ISSUE TITLE
bug(sublime-4): IndexError: tuple index out of range

### ISSUE THREAD

**Issue by Kristinita** [facelessuser/TabsExtra#65]
### 1. Summary

If I use TabsExtra in Sublime Text 4:

    I get traceback in Sublime Text console when I close any tab.

#### 2.1. Safe mode

I reproduced the problem in a Sublime Text Safe mode and in a freshly installed Sublime Text without any settings and packages.

#### 2.2. Steps to reproduce

I installed TabsExtra via Package Control → (optional) I restarted Sublime Text → I opened any file in Sublime Text → I closed this file via standard shortcut (<kbd>Ctrl+W</kbd> in my Windows) → I opened Sublime Text console.

#### 3.1. Expected

No traceback in console.

#### 3.2. Actual

```python
key evt: ctrl + w
command: close
plugin rewritten command: tabs_extra_close
Traceback(most recent call last):
    File "D:\Sublime_Text_build_4143_x64_for_debugging_2\Lib\python33\sublime_plugin.py", line 1456, in run_
        return self.run(**args)
    File "D:\Sublime_Text_build_4143_x64_for_debugging_2\Data\Installed Packages\TabsExtra.sublime-package\tabs_extra.py", line 439, in run
        sublime_api.window_close_file(self.window.id(), v.id())
    File "D:\Sublime_Text_build_4143_x64_for_debugging_2\Lib\python33\sublime_plugin.py", line 883, in on_pre_close
        run_view_callbacks('on_pre_close', view_id)
    File "D:\Sublime_Text_build_4143_x64_for_debugging_2\Lib\python33\sublime_plugin.py", line 708, in run_view_callbacks
        callback(v, *args)
    File "D:\Sublime_Text_build_4143_x64_for_debugging_2\Lib\python33\sublime_plugin.py", line 152, in profiler
        return event_handler(*args)
    File "D:\Sublime_Text_build_4143_x64_for_debugging_2\Data\Installed Packages\TabsExtra.sublime-package\tabs_extra.py", line 538, in on_pre_close
        view.settings().set("tabs_extra_is_closed", True)
    File "D:\Sublime_Text_build_4143_x64_for_debugging_2\Lib\python33\sublime.py", line 1393, in settings
        self.settings_object = Settings(sublime_api.view_settings(self.view_id))
IndexError: tuple index out of range
```

### 4. Environment

1. Microsoft Windows [Version 10.0.19045.2965]
1. Sublime Text Build 4143
1. TabsExtra v. 1.6.1

Thanks.

**Comment by facelessuser** [comment]
Yeah, something is different in later Sublime versions. I don't know when the change occurred, but I'm seeing this as well. I'll have to see if there is something we can do to mitigate this.

What's weird is that we are doing work in `on_pre_close`, but it appears the view index can't be found when updating the view's settings, which makes no sense.

**Comment by facelessuser** [comment]
I've confirmed it is a change to the API. I arbitrarily grabbed Sublime version 4088, and does not use this new callback parameter. So, somewhere between then and now it was added. At some point, the higher-level API also started taking this as an optional parameter while the lower-level API seems to require it. The docs don't state when the API was changed to accept such parameters, so it looks like I'll have to bisect releases until I locate the change 😢.

Anyways, I've at least identified the issue, the rest should be relatively easy once I can pin down the version that introduced this change.
